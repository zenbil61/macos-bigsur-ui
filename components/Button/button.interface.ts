import React from 'react';

interface IButton {
    type?: string;
    width?: number;
    size?: string;
    disabled?: boolean;
    block?: boolean;
    children: React.ReactNode;
    onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

export default IButton;
