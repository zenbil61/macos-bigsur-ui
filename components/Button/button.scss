// @import '../styles/variables';
// @import '../styles/mixins';
@import '../../styles/main';

button {
    border: 0px;
}
.btn {
    cursor: pointer;
    color: white;
    padding: 5px 30px;
    border: 0px;
    @include radius-10;
    @include shadow-4;
    @include bg-color-variant;

    &-sm {
        font-size: 12px;
        padding: 4px 10px;
    }

    &.disabled {
        color: $primary-50;
        @include shade-dark;
    }
}
