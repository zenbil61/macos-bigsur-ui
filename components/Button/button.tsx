import React from "react";
// import "./button.scss";

interface IButton {
  type?: string;
  width?: number;
  size?: string;
  disabled?: boolean;
  block?: boolean;
  children: string;
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
}
const Button: React.FC<IButton> = ({
  type = "primary",
  size,
  disabled,
  block,
  children,
  onClick,
  ...props
}: IButton) => {
  const btnClass = `btn 
    ${type} 
    ${block ? "block" : ""}
    ${disabled ? "disabled" : ""} 
    ${size ? `btn-${size}` : ""}`;

  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
    onClick?.(e);
  };

  return (
    <button
      type="button"
      onClick={handleClick}
      disabled={disabled}
      className={btnClass}
      {...props}
    >
      {children}
    </button>
  );
};

export default Button;
