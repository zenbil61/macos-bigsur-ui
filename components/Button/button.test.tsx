import { render } from '@testing-library/react';
import Button from './button';

test('Button component tests', () => {
    const primaryButton = render(<Button type="primary">Primary Button</Button>);
    const secondaryButton = render(<Button type="secondary">Secondary Button</Button>);
    const successButton = render(<Button type="success">Success Button</Button>);
    const dangerButton = render(<Button type="secondary">Danger Button</Button>);
    const infoButton = render(<Button type="info">Info Button</Button>);
    const warningButton = render(<Button type="warning">Warning Button</Button>);
    const lightButton = render(<Button type="light">Light Button</Button>);

    expect(primaryButton).toMatchSnapshot();
    expect(secondaryButton).toMatchSnapshot();
    expect(successButton).toMatchSnapshot();
    expect(dangerButton).toMatchSnapshot();
    expect(infoButton).toMatchSnapshot();
    expect(warningButton).toMatchSnapshot();
    expect(lightButton).toMatchSnapshot();
});
