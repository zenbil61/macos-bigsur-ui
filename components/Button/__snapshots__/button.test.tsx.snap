// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="btn 
    primary 
    
     
    "
        type="button"
      >
        Primary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Secondary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    success 
    
     
    "
        type="button"
      >
        Success Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Danger Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    info 
    
     
    "
        type="button"
      >
        Info Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    warning 
    
     
    "
        type="button"
      >
        Warning Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    light 
    
     
    "
        type="button"
      >
        Light Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="btn 
    primary 
    
     
    "
      type="button"
    >
      Primary Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button component tests 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="btn 
    primary 
    
     
    "
        type="button"
      >
        Primary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Secondary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    success 
    
     
    "
        type="button"
      >
        Success Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Danger Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    info 
    
     
    "
        type="button"
      >
        Info Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    warning 
    
     
    "
        type="button"
      >
        Warning Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    light 
    
     
    "
        type="button"
      >
        Light Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="btn 
    secondary 
    
     
    "
      type="button"
    >
      Secondary Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button component tests 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="btn 
    primary 
    
     
    "
        type="button"
      >
        Primary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Secondary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    success 
    
     
    "
        type="button"
      >
        Success Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Danger Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    info 
    
     
    "
        type="button"
      >
        Info Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    warning 
    
     
    "
        type="button"
      >
        Warning Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    light 
    
     
    "
        type="button"
      >
        Light Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="btn 
    success 
    
     
    "
      type="button"
    >
      Success Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button component tests 4`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="btn 
    primary 
    
     
    "
        type="button"
      >
        Primary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Secondary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    success 
    
     
    "
        type="button"
      >
        Success Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Danger Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    info 
    
     
    "
        type="button"
      >
        Info Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    warning 
    
     
    "
        type="button"
      >
        Warning Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    light 
    
     
    "
        type="button"
      >
        Light Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="btn 
    secondary 
    
     
    "
      type="button"
    >
      Danger Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button component tests 5`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="btn 
    primary 
    
     
    "
        type="button"
      >
        Primary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Secondary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    success 
    
     
    "
        type="button"
      >
        Success Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Danger Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    info 
    
     
    "
        type="button"
      >
        Info Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    warning 
    
     
    "
        type="button"
      >
        Warning Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    light 
    
     
    "
        type="button"
      >
        Light Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="btn 
    info 
    
     
    "
      type="button"
    >
      Info Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button component tests 6`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="btn 
    primary 
    
     
    "
        type="button"
      >
        Primary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Secondary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    success 
    
     
    "
        type="button"
      >
        Success Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Danger Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    info 
    
     
    "
        type="button"
      >
        Info Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    warning 
    
     
    "
        type="button"
      >
        Warning Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    light 
    
     
    "
        type="button"
      >
        Light Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="btn 
    warning 
    
     
    "
      type="button"
    >
      Warning Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button component tests 7`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="btn 
    primary 
    
     
    "
        type="button"
      >
        Primary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Secondary Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    success 
    
     
    "
        type="button"
      >
        Success Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    secondary 
    
     
    "
        type="button"
      >
        Danger Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    info 
    
     
    "
        type="button"
      >
        Info Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    warning 
    
     
    "
        type="button"
      >
        Warning Button
      </button>
    </div>
    <div>
      <button
        class="btn 
    light 
    
     
    "
        type="button"
      >
        Light Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="btn 
    light 
    
     
    "
      type="button"
    >
      Light Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
